mmv1/third_party/terraform/tpgresource/field_helpers.go

```diff
@@ -383,7 +383,8 @@ func GetRegionFromSchema(regionSchemaField, zoneSchemaField string, d TerraformR
 		return GetResourceNameFromSelfLink(v.(string)), nil
 	}
 	if v, ok := d.GetOk(zoneSchemaField); ok && zoneSchemaField != "" {
-		return GetRegionFromZone(v.(string)), nil
+		zoneSelfLink := GetRegionFromZone(v.(string))
+		return GetResourceNameFromSelfLink(zoneSelfLink), nil
```

Comment:
(nit)

For this code currently these are the types of values found at different stages:

| Start value | Value of `zoneSelfLink` | End value | 
|-|-|-|
|https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1-a | https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1 | us-central1 |
| us-central1-a | us-central1 | us-central1 |

What do you think about renaming the variable or swapping round the two operations in this code block?

Reply:
I see what you mean! It makes more sense to swap the two operations. Good catch!